### PR TITLE
multi: update ChannelManager tx broadcaster from Arc to Deref

### PR DESCRIPTION
**Pull request by valentinewallace**
Todo: should break up into multiple commits for reviewability (even though they'll have to be squashed in the end for git bisect reasons).

It's basically the same as #443 though, mostly mechanical changes.

Partially addresses #237.

### CODE REVIEW

File: lightning/src/ln/channelmanager.rs [hunk 1]
```diff
@@ -289,15 +289,15 @@ const ERR: () = "You need at least 32 bit pointers (well, usize, but we'll assum
...
+pub type SimpleArcChannelManager<M, T> = Arc<ChannelManager<InMemoryChannelKeys, Arc<M>, Arc<T>>>;
 
 /// SimpleRefChannelManager is a type alias for a ChannelManager reference, and is the reference
 /// counterpart to the SimpleArcChannelManager type alias. Use this type by default when you don't
 /// need a ChannelManager with a static lifetime. You'll need a static lifetime in cases such as
 /// usage of lightning-net-tokio (since tokio::spawn requires parameters with static lifetimes).
 /// But if this is not necessary, using a reference is more efficient. Defining these type aliases
 /// helps with issues such as long function definitions.
-pub type SimpleRefChannelManager<'a, M> = ChannelManager<InMemoryChannelKeys, &'a M>;
+pub type SimpleRefChannelManager<'a, M, T> = ChannelManager<InMemoryChannelKeys, &'a M, &'a T>;
```
Comment (TheBlueMatt) [inline lightning/src/ln/channelmanager.rs]: Hmm, shouldn't this get a 'b as well?

Comment (valentinewallace) [reply lightning/src/ln/channelmanager.rs]: True, fixed

File: lightning/src/ln/functional_test_utils.rs [hunk 2]
```diff
@@ -908,43 +912,53 @@ pub fn fail_payment_along_route<'a, 'b>(origin_node: &Node<'a, 'b>, expected_rou
...
+		let node = ChannelManager::new(Network::Testnet, cfgs[i].fee_estimator.clone(), &cfgs[i].chan_monitor, cfgs[i].tx_broadcaster, cfgs[i].logger.clone(), cfgs[i].keys_manager.clone(), if node_config[i].is_some() { node_config[i].clone().unwrap() } else { default_config }, 0).unwrap();
 		chanmgrs.push(node);
 	}
 
 	chanmgrs
 }
 
-pub fn create_network<'a, 'b>(node_count: usize, cfgs: &'a Vec<NodeCfg>, chan_mgrs: &'b Vec<ChannelManager<EnforcingChannelKeys, &'a TestChannelMonitor>>) -> Vec<Node<'a, 'b>> {
+pub fn create_network<'a, 'b: 'a, 'c: 'b>(node_count: usize, cfgs: &'b Vec<NodeCfg<'c>>, chan_mgrs: &'a Vec<ChannelManager<EnforcingChannelKeys, &'b TestChannelMonitor<'c>, &'c test_utils::TestBroadcaster>>) -> Vec<Node<'a, 'b, 'c>> {
+// pub fn create_network<'a, 'b, 'c>(node_count: usize, cfgs: &'b Vec<NodeCfg<'c>>, chan_mgrs: &'a Vec<ChannelManager<EnforcingChannelKeys, &'b TestChannelMonitor<'c>, &'c test_utils::TestBroadcaster>>) -> Vec<Node<'a, 'b, 'c>> {
```
Comment (TheBlueMatt) [inline lightning/src/ln/functional_test_utils.rs]: oops'd

**Comment by valentinewallace** [comment]
> Looks good. One stray nit and good to go.

Done :)

**Pull request merged**